### PR TITLE
Activity log: Migrate activity data to QueryManager

### PR DESCRIPTION
**Pull request by sirreal**
Visit https://calypso.live/stats/activity?branch=update/activity-log/use-query-manager

Includes #17392

### CODE REVIEW

File: client/lib/query-manager/activity/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ */
+import QueryManager from 'lib/query-manager';
+
+/**
+ * ActivityQueryManager manages Activity which can be queried
+ */
+export default class ActivityQueryManager extends QueryManager {
+	constructor( data, options ) {
+		super( data, {
+			itemKey: 'ts_utc',
```
Comment (ockham) [inline client/lib/query-manager/activity/index.js]: For `ThemeQueryManager`, we also have to set `itemKey`, but we do it at consumer (i.e. reducer) level, https://github.com/Automattic/wp-calypso/blob/8bc7c784d354614b352beb2d05940bcde291dfa7/client/state/themes/reducer.js#L276

(Fortunately, there's only one consumer.)

I'm not sure which is preferable. Conceptually, hard-coding that inside the class definition seems good, but OTOH, not having to do that is alluring, too.

File: client/state/selectors/get-activity-logs.js [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
+const getActivityLogs = createSelector(
+	( state, siteId ) => {
+		const manager = get( state, [ 'activityLog', 'logItems', siteId ], null );
+		if ( ! manager ) {
+			return null;
+		}
+
+		return manager.getItems();
+	},
+	( state, siteId ) => get( state, [ 'activityLog', 'logItems', siteId ] )
```
Comment (sirreal) [inline client/state/selectors/get-activity-logs.js]: State is not updating, there's likely a problem here.

File: client/state/activity-log/log/reducer.js [hunk 3]
```diff
@@ -1,18 +1,34 @@
...
 
+// FIXME: Restore persistence after implementing queryManager
+//import { logItemsSchema } from './schema';
+
+const debug = debugFactory( 'calypso:state:activity-log:reducer' );
+
 // FIXME: No-op reducers
 export const logError = keyedReducer( 'siteId', state => state );
 
 export const logItems = keyedReducer(
```
Comment (sirreal) [inline client/state/activity-log/log/reducer.js]: I wonder if `keyedReducer` plays well with `QueryManager` 🤔

Comment (sirreal) [inline client/state/activity-log/log/reducer.js]: Confirmed. `QueryManager` mutates and does not play well with `keyedReducer`:

https://github.com/Automattic/wp-calypso/blob/0906108b313fefd9bae151bee64c3f953159c811/client/state/utils.js#L114-L117

Comment (ockham) [inline client/state/activity-log/log/reducer.js]: Good sleuthing! Hopefully we can fix this at a generic level in order to allow `QM` too be used with `keyedReducer`.

Comment (sirreal) [inline client/state/activity-log/log/reducer.js]: This was a false start, `QM` does appear to be a good redux state citizen (immutable).

Comment (sirreal) [inline client/state/activity-log/log/reducer.js]: Found it 🤦‍♂️ e6c622011a149f3f405c0a664c90e642592b1c66

The constructor and `receive` methods have slightly different signatures.

Comment (ockham) [inline client/state/activity-log/log/reducer.js]: I might've bitten by that before... :slightly_frowning_face:

File: client/my-sites/stats/activity-log-day/index.jsx [hunk 4]
```diff
@@ -101,35 +89,32 @@ class ActivityLogDay extends Component {
...
+								args: formattedDate,
+								comment: 'Long date with today indicator, i.e. "January 1, 2017 — Today"',
+							} )
+						: formattedDate }
+					{ /* eslint-enable indent */ }
+				</div>
+				<div className="activity-log-day__events">
+					{ /* eslint-disable indent */ }
+					{ isEmpty( logs )
+						? translate( 'No Events' )
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-day/index.jsx]: Hi! I've found a possible matching string that has already been translated **30** times:
`translate( 'No upcoming events' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/activity-log/use-query-manager)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by sirreal** [comment]
Superseded by #17135. This PR is outdated and the rebase was becoming complicated. Instead I decided to cherry-pick to a new branch.

**Pull request closed without merging**